common/finalizer/finalizer.go

```diff
@@ -144,7 +144,12 @@ func (f *Finalizer) Run(
 
 	completionChannel := make(chan struct{})
 	go func() {
+		// prevent holding on to the callbacks for longer than needed and allow garbage collection
+		// (safe since any calls to Register/Deregister will be aborted now that the finalizer ran)
+		defer func() { f.callbacks = nil }()
+
```

Comment:
I think there's a potential memory leak here that this addresses:

say, the finalizer has a few registered callbacks
each callback is a closure that references the cache item, which itself references the finalizer (see cache.go)
once a finalizer is executed, it is "sealed" off in that it doesn't allow to add/remove any callbacks anymore
the issue is that cache items that are evicted from the cache cannot remove themselves from the (now "sealed") finalizer anymore
therefore, until all cache items that are registered with the finalizer are evicted, the finalizer cannot be garbage collected
this will likely happen eventually, but by clearing the callbacks after the run, the process is expedited
(another fix I had considered was to have the callbacks capture the cache key instead of the cache item - breaking the cycle apart - but this approach here seems sufficient and easier)